app/uk/gov/hmrc/gform/gform/EnrolmentController.scala

```diff
@@ -215,10 +216,18 @@ class EnrolmentController(
           idenVer <- extractIdentifiersAndVerifiers[F](enrolmentSection)
           (identifierss, verifiers) = idenVer
           identifiers = identifierss.map(_._2)
-          _       <- validateIdentifiers[F](identifierss, postCheck)
-          _       <- enrolmentService.enrolUser(serviceId, identifiers, verifiers, retrievals)
-          authRes <- checkEnrolment(identifiers)
-        } yield authRes
+          _             <- validateIdentifiers[F](identifierss, postCheck)
+          _             <- enrolmentService.enrolUser(serviceId, identifiers, verifiers, retrievals)
+          initialResult <- checkEnrolment(identifiers)
+
+          reattemptResult <- (initialResult, legacyFcEnrolmentVerifier) match {
+                              case (EnrolmentFailed, Some(lfcev)) =>
+                                enrolmentService
+                                  .enrolUser(serviceId, identifiers, List(Verifier(lfcev.value, "FC")), retrievals)
+                                checkEnrolment(identifiers)
```

Comment:
Just a side note. This is race condition as line 225 and 227 will run concurrently since they are not part of single for comprehension. 

Let's make a function `f` which will take `verifiers` as an input and let's reuse it instead. Something like:
```
def f(verifiers: List[Verifier]) = for {
  _      <- enrolmentService.enrolUser(serviceId, identifiers, verifiers, retrievals)
  result <- checkEnrolment(identifiers)
} yield result
```